staticd/static_vty.c

```diff
@@ -73,10 +73,25 @@ static int static_route_leak(struct vty *vty, const char *svrf,
 	route_tag_t tag = 0;
 	uint32_t table_id = 0;
 	const struct lyd_node *dnode;
+	const char *flag = flag_str;
+	const char *ifp = ifname;
 
 	memset(buf_src_prefix, 0, PREFIX_STRLEN);
 	memset(buf_nh_type, 0, PREFIX_STRLEN);
 
+	if (ifname) {
```

Comment:
I am not sure about this.

Currently, if you try to use some parameters that are not allowed for `blackhole` and `reject` routes, for example, `onlink` or `nexthop-vrf`, you receive the following error: `Nexthop interface name can not be from reserved keywords(Null0, reject, blackhole)`. This doesn't work for `Null0` routes, because of this hack with `ifname` and `flag_str` – `Null0` is being treated as a regular route instead of a blackhole route.

With this change, you actually expand this problem with `Null0` to also `blackhole` and `reject` routes. Please, check my PR https://github.com/FRRouting/frr/pull/9006 - I think this is the right way to fix this.